cylc/flow/scheduler.py

```diff
@@ -1846,18 +1847,16 @@ def _update_workflow_state(self):
 
     async def update_data_structure(self, reloaded: bool = False):
         """Update DB, UIS, Summary data elements"""
-        # Add tasks that have moved from runahead to live pool.
-        if self.data_store_mgr.updates_pending:
-            # Collect/apply data store updates/deltas
-            self.data_store_mgr.update_data_structure(reloaded=reloaded)
-            # Publish updates:
-            if self.data_store_mgr.publish_pending:
-                self.data_store_mgr.publish_pending = False
-                self.server.publish_queue.put(
-                    self.data_store_mgr.publish_deltas)
-                # Non-async sleep - yield to other threads rather
-                # than event loop
-                sleep(0)
+        # Collect/apply data store updates/deltas
+        self.data_store_mgr.update_data_structure(reloaded=reloaded)
+        # Publish updates:
+        if self.data_store_mgr.publish_pending:
+            self.data_store_mgr.publish_pending = False
+            self.server.publish_queue.put(
+                self.data_store_mgr.publish_deltas)
+            # Non-async sleep - yield to other threads rather
+            # than event loop
+            sleep(0)
         # Database update
         self.workflow_db_mgr.put_task_pool(self.pool)
         self.update_data_store()
```

Comment:
There might be an issue here... that method sets `self.data_store_mgr.updates_pending` to `True` .. Which seems kinda circular if there's no criteria other than `update_data_structure` running..

Reply:
I'm not sure I understand - what symptoms of this might I see? Are you saying that I should keep `if self.data_store_mgr.updates_pending:` here too, or that the way it was before my change also seems sketchy? If so, could you have a look into it.

Reply:
Because `self.update_data_store()` sets `self.data_store_mgr.updates_pending` to `True`, then:
```
 if has_updated or self.data_store_mgr.updates_pending:
```
will run `update_data_structure` every main loop now including the DB update (which isn't a data-store thing)..

Reply:
(actually not sure why `self.update_data_store()` is even called in `update_data_structure`)

Reply:
> will run update_data_structure every main loop now

Yikes!

Reply:
@hjoliver  - do you know why you added `self.update_data_store` in #5231 [link to diff](https://github.com/cylc/cylc-flow/commit/4a6b447fdbd4c28e6bdecaf3814a4d4cdda9652b#r123264068)? 

I don't think there's any need to call it, but I'm going to try some in-practice tests with it locally.